### PR TITLE
BS-14270, fix document init with contract input

### PR DESCRIPTION
**Pull request by baptistemesta**
The initialization of process instance with contract input was not good due to wrong Expression context used

Regression was introduces by the fix on Business Data initialization

Cleaned up the start of process to always have the good expression context: in case of call activity use expression context of call activity only for operation execution

### CODE REVIEW

File: bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreator.java [hunk 1]
```diff
@@ -546,72 +545,66 @@ public void createConnectorInstances(final PersistentObject container, final Lis
         final List<SConnectorInstance> connectorInstances = new ArrayList<SConnectorInstance>(connectors.size());
         int executionOrder = 0;
         for (final SConnectorDefinition sConnectorDefinition : connectors) {
-            connectorInstances.add(BuilderFactory
-                    .get(SConnectorInstanceBuilderFactory.class)
-                    .createNewInstance(sConnectorDefinition.getName(), container.getId(), containerType,
-                            sConnectorDefinition.getConnectorId(), sConnectorDefinition.getVersion(), sConnectorDefinition.getActivationEvent(),
-                            executionOrder++)
-                    .done());
+            final int executionOrder1 = executionOrder++;
```
Comment (educhastenier) [inline bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreator.java]: `executionOrder++` increments after use, which changes the previous behaviour. Are u sure this is what you want?

Comment (baptistemesta) [inline bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreator.java]: no the previous behavior is the same, first call to createConnectorInstanceObject is with 0, second call with 1, 
Before the first create new instance was called with 0 first then with 1. there is an UT on this

File: bonita-integration-tests/bonita-integration-tests-client/src/main/java/org/bonitasoft/engine/activity/ContractIT.java [hunk 2]
```diff
@@ -192,17 +192,22 @@ public void should_getUserTaskContract_return_the_contract() throws Exception {
...
                 .addInput("amount", Type.DECIMAL, "expense amount")
                 .addComplexInput("date", "expense date").addInput("expenseType", Type.TEXT, "describe expense type");
         contractDefinitionBuilder.addFileInput("report", "myReport");
+        contractDefinitionBuilder = builder.addContract();
+        contractDefinitionBuilder.addFileInput("reportInit", "myReport");
         //when
         final ProcessDefinition processDefinition = deployAndEnableProcessWithActor(builder.done(), ACTOR_NAME, matti);
-        getProcessAPI().startProcess(processDefinition.getId());
+        getProcessAPI().startProcessWithInputs(processDefinition.getId(),
+                Collections.<String, Serializable> singletonMap("reportInit", new FileInputValue("theFile", "theContent".getBytes())));
```
Comment (educhastenier) [inline bonita-integration-tests/bonita-integration-tests-client/src/main/java/org/bonitasoft/engine/activity/ContractIT.java]: Missing assert on the initialized document

File: bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreator.java (hunk 1 above)
Comment (educhastenier) [inline bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreator.java]: Inline it.

File: bpm/bonita-core/bonita-process-engine/src/test/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreatorTest.java [hunk 3]
```diff
@@ -61,35 +58,25 @@
...
 
         bpmInstancesCreator.createConnectorInstances(container, connectors, SConnectorInstance.FLOWNODE_TYPE);
-        verify(connectorBuilderFact, times(1)).createNewInstance(anyString(), anyLong(), anyString(), anyString(), anyString(), any(ConnectorEvent.class),
-                eq(0));
-        verify(connectorBuilderFact, times(1)).createNewInstance(anyString(), anyLong(), anyString(), anyString(), anyString(), any(ConnectorEvent.class),
-                eq(1));
+
+        verify(bpmInstancesCreator).createConnectorInstanceObject(any(PersistentObject.class), anyString(), any(SConnectorDefinition.class),eq(0));
+        verify(bpmInstancesCreator).createConnectorInstanceObject(any(PersistentObject.class), anyString(), any(SConnectorDefinition.class),eq(1));
         ignoreStubs(transactionExecutor);
```
Comment (educhastenier) [inline bpm/bonita-core/bonita-process-engine/src/test/java/org/bonitasoft/engine/bpm/model/impl/BPMInstancesCreatorTest.java]: Is it still necessary?

**Pull request merged**